plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileFormats.java

```diff
@@ -616,10 +616,11 @@ public void testParquetPageSourceSchemaEvolution(int rowCount)
     public void testParquetCaseSensitivity(int rowCount)
             throws Exception
     {
-        TestColumn writeColumn = new TestColumn("UPPER_CASE_COLUMN", createVarcharType(4), new HiveVarchar("test", 4), utf8Slice("test"));
-        TestColumn readColumn = new TestColumn("uppeR_casE_columN", createVarcharType(4), new HiveVarchar("test", 4), utf8Slice("test"));
+        TestColumn writeColumnA = new TestColumn("UPPER_CASE_COLUMN", createVarcharType(5), new HiveVarchar("testA", 5), utf8Slice("testA"));
+        TestColumn writeColumnB = new TestColumn("Upper_Case_Column", createVarcharType(5), new HiveVarchar("testB", 5), utf8Slice("testB"));
+        TestColumn readColumn = new TestColumn("uppeR_casE_columN", createVarcharType(5), new HiveVarchar("testA", 5), utf8Slice("testA"));
```

Comment:
The new test fails even before https://github.com/trinodb/trino/commit/9a32c1941104b8b2abdb71809c34d386e2e9eaeb commit. I think this test doesn't cover the reported issue. Probably, it was a mixed problem of field case sensitivity and projections.

Reply:
Yes, there was a failure in `io.trino.parquet.predicate.PredicateUtils#getStatistics` before. However, that failure would only manifest if the duplicated column is selected from file while the failure in `createPrunedColumnsMetadata` happened regardless of query.
In any case, both the scenarios should not fail now.

Reply:
I'm confused why using the first case-insensitive match is okay. I believe we usually throw an exception for such ambiguous cases in this project.

Reply:
There wasn't any intentional handling or test cases for this scenario before. And we know from customer reports that at least some cases with this ambiguity used to work before. So resolving the query failures by deterministically picking some match is the only pragmatic solution here. The alternative of asking users to rewrite their files is too disruptive.